jaxtronomy/LensModel/Profiles/sis.py

```diff
@@ -0,0 +1,175 @@
+__author__ = "sibirrer"
+
+import jax
+import jax.numpy as jnp
+from lenstronomy.LensModel.Profiles.base_profile import LensProfileBase
+
+jax.config.update("jax_enable_x64", True)  # 64-bit floats
+
+__all__ = ["SIS"]
+
+
+class SIS(LensProfileBase):
+    """This class contains the function and the derivatives of the Singular Isothermal
+    Sphere.
+
+    .. math::
+        \\kappa(x, y) = \\frac{1}{2} \\left(\\frac{\\theta_{E}}{\\sqrt{x^2 + y^2}} \\right)
+
+    with :math:`\\theta_{E}` is the Einstein radius,
+    """
+
+    param_names = ["theta_E", "center_x", "center_y"]
+    lower_limit_default = {"theta_E": 0, "center_x": -100, "center_y": -100}
+    upper_limit_default = {"theta_E": 100, "center_x": 100, "center_y": 100}
+
+    def function(self, x, y, theta_E, center_x=0, center_y=0):
+        x_shift = x - center_x
+        y_shift = y - center_y
+        f_ = theta_E * jnp.sqrt(x_shift * x_shift + y_shift * y_shift)
+        return f_
+
+    def derivatives(self, x, y, theta_E, center_x=0, center_y=0):
+        """Returns df/dx and df/dy of the function."""
+        x_shift = x - center_x
+        y_shift = y - center_y
+        R = jnp.sqrt(x_shift * x_shift + y_shift * y_shift)
+        if isinstance(R, int) or isinstance(R, float):
+            a = theta_E / max(0.000001, R)
+        else:
+            a = jnp.empty_like(R)
+            r = R[R > 0]  # in the SIS regime
+            # instead of x[idx] = y, use x = x.at[idx].set(y)
+            a = a.at[R == 0].set(0)
+            a = a.at[R > 0].set(theta_E / r)
+        f_x = a * x_shift
+        f_y = a * y_shift
+        return f_x, f_y
+
+    def hessian(self, x, y, theta_E, center_x=0, center_y=0):
+        """Returns Hessian matrix of function d^2f/dx^2, d^2/dxdy, d^2/dydx,
+        d^f/dy^2."""
+        x_shift = x - center_x
+        y_shift = y - center_y
+        R = (x_shift * x_shift + y_shift * y_shift) ** (3.0 / 2)
+        if isinstance(R, int) or isinstance(R, float):
+            prefac = theta_E / max(0.000001, R)
```

Comment:
would be good having this tested to make sure it is JAX compatible (or else find another way to deal with R=0

Reply:
Thanks for the tip @sibirrer, I removed the `isinstance` in favour of `jnp.where` . The tests now cover 100% of the profile.